examples/z_ping.c

```diff
@@ -0,0 +1,118 @@
+#include <pthread.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+
+#include "zenoh.h"
+
+pthread_cond_t cond;
+pthread_mutex_t mutex;
+
+void callback(const z_sample_t* sample, void* context) { pthread_cond_signal(&cond); }
+void drop(void* context) { pthread_cond_destroy(&cond); }
+
+struct args_t {
+    unsigned int size;             // -s
+    unsigned int number_of_pings;  // -n
+    unsigned int warmup_ms;        // -w
+    char* config_path;             // -c
+    uint8_t help_requested;        // -h
+};
+struct args_t parse_args(int argc, char** argv);
+
+int main(int argc, char** argv) {
+    struct args_t args = parse_args(argc, argv);
+    if (args.help_requested) {
+        printf(
+            "\
+		-n (optional, int, default=4): the number of pings to be attempted\n\
+		-s (optional, int, default=8): the size of the payload embedded in the ping and repeated by the pong\n\
+		-w (optional, int, default=0): the warmup time in ms during which pings will be emitted but not measured\n\
+		-c (optional, string): the path to a configuration file for the session. If this option isn't passed, the default configuration will be used.\n\
+		");
+        return 1;
+    }
+    pthread_mutex_init(&mutex, NULL);
+    pthread_cond_init(&cond, NULL);
+    z_owned_config_t config = args.config_path ? zc_config_from_file(args.config_path) : z_config_default();
+    z_owned_session_t session = z_open(z_move(config));
+    z_keyexpr_t ping = z_keyexpr_unchecked("test/ping");
+    z_keyexpr_t pong = z_keyexpr_unchecked("test/pong");
+    z_owned_publisher_t pub = z_declare_publisher(z_loan(session), ping, NULL);
+    z_owned_closure_sample_t respond = z_closure(callback, drop, (void*)(&pub));
+    z_owned_subscriber_t sub = z_declare_subscriber(z_loan(session), pong, z_move(respond), NULL);
+    uint8_t* data = malloc(args.size);
+    for (int i = 0; i < args.size; i++) {
+        data[i] = i % 10;
+    }
+    pthread_mutex_lock(&mutex);
+    if (args.warmup_ms) {
+        printf("Warming up for %dms...\n", args.warmup_ms);
+        clock_t warmup_end = clock() + CLOCKS_PER_SEC * args.warmup_ms / 1000;
+        for (clock_t now = clock(); now < warmup_end; now = clock()) {
+            z_publisher_put(z_loan(pub), data, args.size, NULL);
+            pthread_cond_wait(&cond, &mutex);
+        }
+    }
+    clock_t* results = malloc(sizeof(clock_t) * args.number_of_pings);
+    for (int i = 0; i < args.number_of_pings; i++) {
+        clock_t start = clock();
+        z_publisher_put(z_loan(pub), data, args.size, NULL);
+        int err = pthread_cond_wait(&cond, &mutex);
```

Comment:
The `err` variable is never checked.

Reply:
I forgot to remove that after checking it with prints earlier, removing the assignment x)